setup/src/Magento/Setup/Console/Command/AdminUserCreateCommand.php

```diff
@@ -62,7 +75,9 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return \Magento\Framework\Console\Cli::RETURN_FAILURE;
         }
         $installer = $this->installerFactory->create(new ConsoleLogger($output));
-        $installer->installAdminUser($input->getOptions());
+        $options = $input->getOptions();
+        $options['db-prefix'] = $this->deploymentConfig->get(ConfigOptionsListConstants::CONFIG_PATH_DB_PREFIX);
+        $installer->installAdminUser($options);
```

Comment:
Fix must be moved into model `\Magento\Setup\Model\Installer::installAdminUser`. This class is already aware of `deploymentConfig`, all occurrences of `DB_PREFIX` are already in models.

Reply:
Added `Magento\Setup\Module\Setup` to Interface `AdminAccountFactory` inside installAdminUser.

I hope these changes are ok for you.

Reply:
I don't think we should change `AdapterInterface` back to `Setup`.

I suggested just to move DB prefix obtaining from command to `\Magento\Setup\Model\Installer::installAdminUser` model (not to any other model).